extensions/amp-intersection-observer-polyfill/0.1/cross-origin-observer.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MessageType} from '../../../src/3p-frame-messaging';
+import {iframeMessagingClientFor} from '../../../src/inabox/inabox-iframe-messaging-client';
+import {layoutRectLtwh} from '../../../src/layout-rect';
+
+/**
+ * Setup cross-origin iframe polyfill for AMP iframes, such as inabox.
+ *
+ * @param {!Window} win
+ */
+export function maybeSetupCrossOriginObserver(win) {
+  // Cross-origin iframe polyfill for AMP iframes, such as inabox.
+  const iframeClient = iframeMessagingClientFor(win);
```

Comment:
`iframeClient` is the check to setup cross origin observer. Should we rely on `getMode.(win).runtime` instead to be more specific?

Reply:
Sure. I can add it as well. Though I thought technically, if `iframeClient` is available - it doesn't matter what type of runtime we are in.

Reply:
Just realized this is not always the case. inabox can be served in friendly iframe as well. 
https://github.com/ampproject/amphtml/blob/c619be8c5751243be74df8847cfb176b6e20872d/src/inabox/inabox-viewport.js#L481-L483

I believe we need to check against `isFriendlyIframed_` as well.

Reply:
This code is only be executing when loaded in a specific context - the iframe. And only when iframeClient is also set up. I think at least one of these should always be `false` for a friendly iframe case. No?

Reply:
For instance, at least [here](https://github.com/ampproject/amphtml/blob/90f1f7fcff01c68d703140e1cb6f6da2ebbabbef/src/inabox/inabox-iframe-messaging-client.js#L40) I see that iframe client is not setup for a friendly case. If my reading of this is right.

Reply:
Ohhh, I didn't know that the iframe client is only set up when in friendly iframe. You're absolute right about checking `iframeClient` instead of `getMode(win).runtime`! LGTM